File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/3037.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

n = parseInt(lines.shift())
for(i=0;i<n;i++){
    maria = 0;
    joao = 0;
    for(j=0;j<3;j++){
        ls = lines.shift().split(' ')
        x = parseInt(ls[0])
        d = parseInt(ls[1])
        joao+=x*d
    }
    for(j=3;j<6;j++){
        ls = lines.shift().split(' ')
        x = parseInt(ls[0])
        d = parseInt(ls[1])
        maria+=x*d
    }
    if(joao>maria){
        console.log("JOAO")
    } else{
        console.log("MARIA")
    }
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/3037.py
n = int(input())
for i in range(0, n):
    maria = 0
    joao = 0
    for j in range(0, 3):
        ls = input().split(' ')
        x = int(ls[0])
        d = int(ls[1])
        joao=joao+x*d
    
    for j in range(3, 6):
        ls = input().split(' ')
        x = int(ls[0])
        d = int(ls[1])
        maria=maria+x*d
    
    if(joao>maria):
        print("JOAO")
    else:
        print("MARIA")
    
